test-util/src/main/java/org/triplea/test/lib/MatchBuilder.java

```diff
@@ -0,0 +1,60 @@
+package org.triplea.test.lib;
+
+import java.util.function.Function;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+
+import lombok.Builder;
+import lombok.NonNull;
+
+/**
+ * Convenience class for building a type safe hamcrest matcher.
+ * See Test cases for more example usages.
+ * Example usage:
+ * 
+ * <pre>
+ * {@code
+ *  assertThat(testValue, customMatcher("valueToMatch"));
+ *
+ *  private static Matcher<String> customMatcher(final String expectedValue) {
+ *    return MatchBuilder.<String>builder()
+ *        .description("We expect the test value to be equal to: " + expectedValue)
+ *        .checkCondition(String::equals)
+ *        .debug(testValue -> "This can be used to print more debug output when there is a failure"
+ *           + ", it is optional, when omitted you will get a 'testValue.toString()' by default")
+ *        .build();
+ *  }
+ * }
+ * </pre>
+ *
+ * @param <T> The result type which we will be checking. For example, if we are testing a 'HashMap' has a specific
+ *        value, then this would be of type 'HashMap'
+ */
+@Builder
+public class MatchBuilder<T> extends BaseMatcher<T> {
```

Comment:
Having the word "builder" repeat in an expression like `MatchBuilder.builder()` reads strange.  Could you not name this class just `Match`?

Actually, `Matcher` is probably more appropriate given that the name of the Hamcrest class is `Matcher` not `Match`.  If having to fully-qualify the Hamcrest type to avoid a name collision smells, then maybe calling it `CustomMatcher`is a good compromise.

Reply:
Good suggestions, IMO it's between: `CustomMatcher.builder()` or `Matchers.builder()`

The latter suffers from name conflicts as well, NBD but still not ideal.  Will go with former: 984465e